src/hash.hpp

```diff
@@ -13,6 +13,8 @@
 #include "common.h"
 #include "cyclic.hpp"
 
+#include <cstring>
```

Comment:
Was this include actually necessary? I don't see you depending on any `str*` functions.